### PR TITLE
Improved stability of smiles renderer

### PR DESCRIPTION
**Pull request by cpignedoli**
We use now pybel that seems to work for more cases.
Also the search among conformers was removed being too slow

### CODE REVIEW

**Comment by yakutovicha** [comment]
@cpignedoli, before I start reviewing can you please fix all the pre-commit issues?

File: aiidalab_widgets_base/structures.py [hunk 1]
```diff
@@ -538,13 +541,16 @@ def __init__(self, title=''):
...
-        return asemol
+        species = [chemical_symbols[atm.atomicnum] for atm in pymol.atoms]
+        pos = np.asarray([atm.coords for atm in pymol.atoms])
+        pca = PCA(n_components=3)
+        posnew = pca.fit_transform(pos)
+        atoms = Atoms(species, positions=posnew)
+        sys_size = np.ptp(atoms.positions, axis=0)
+        atoms.pbc = True
+        atoms.cell = sys_size + 10
+        atoms.center()
```
Comment (yakutovicha) [inline aiidalab_widgets_base/structures.py:552]:
```suggestion
        atoms = Atoms(species, positions=posnew, pbc=True, cell=np.ptp(posnew, axis=0)+10)
        atoms.center()
```

Comment (yakutovicha) [reply aiidalab_widgets_base/structures.py:552]: Not 100% sure it will work, please try.

File: aiidalab_widgets_base/structures.py [hunk 2]
```diff
@@ -555,17 +561,19 @@ def _optimize_mol(self, mol):
...
 
         # initial cleanup before the weighted search
-        f_f.SteepestDescent(5500, 1.0e-9)
-        f_f.WeightedRotorSearch(15000, 500)
-        f_f.ConjugateGradients(6500, 1.0e-10)
+        f_f.Setup(mol.OBMol)
+        f_f.SteepestDescent(5000, 1.0e-9)
+        #f_f.SteepestDescent(5500, 1.0e-9)
+        #f_f.WeightedRotorSearch(15000, 500)
+        #f_f.ConjugateGradients(6500, 1.0e-9)
```
Comment (yakutovicha) [inline aiidalab_widgets_base/structures.py:576]:
```suggestion
```

File: aiidalab_widgets_base/structures.py [hunk 3]
```diff
@@ -555,17 +561,19 @@ def _optimize_mol(self, mol):
...
-        f_f = pybel._forcefields["mmff94"]  # pylint: disable=protected-access
+        f_f = pybel._forcefields["uff"]  # pylint: disable=protected-access
         if not f_f.Setup(mol.OBMol):
-            f_f = pybel._forcefields["uff"]  # pylint: disable=protected-access
+            f_f = pybel._forcefields["mmff94"]  # pylint: disable=protected-access
             if not f_f.Setup(mol.OBMol):
                 self.output.value = "Cannot set up forcefield"
                 return
 
         # initial cleanup before the weighted search
```
Comment (yakutovicha) [inline aiidalab_widgets_base/structures.py:571]:
```suggestion
        # Initial cleanup before the weighted search.
```

**Review by yakutovicha — CHANGES_REQUESTED** [review]
Thanks @cpignedoli, a few suggestions/questions.

File: aiidalab_widgets_base/structures.py [hunk 4]
```diff
@@ -538,13 +541,16 @@ def __init__(self, title=''):
...
-        asemol = Atoms()
-        for atm in pymol.atoms:
-            asemol.append(Atom(chemical_symbols[atm.atomicnum], atm.coords))
-        asemol.cell = np.amax(asemol.positions, axis=0) - np.amin(asemol.positions, axis=0) + [10] * 3
-        asemol.pbc = True
-        asemol.center()
-        return asemol
+        species = [chemical_symbols[atm.atomicnum] for atm in pymol.atoms]
+        pos = np.asarray([atm.coords for atm in pymol.atoms])
+        pca = PCA(n_components=3)
```
[the remark this answers is not in the document]
Comment (cpignedoli) [reply aiidalab_widgets_base/structures.py:545]: there is no "return asemol" in the file that I pushed for PR teh funcion that i pushed is:
    @staticmethod
    def pymol_2_ase(pymol):
        """Convert pymol object into ASE Atoms."""

        species = [chemical_symbols[atm.atomicnum] for atm in pymol.atoms]
        pos = np.asarray([atm.coords for atm in pymol.atoms])
        pca = PCA(n_components=3)
        posnew = pca.fit_transform(pos)
        atoms = Atoms(species, positions=posnew)
        sys_size = np.ptp(atoms.positions, axis=0)
        atoms.pbc = True
        atoms.cell = sys_size + 10
        atoms.center()
        return atoms

**Comment by cpignedoli** [comment]
I checked, the modifications work

**Review by yakutovicha — APPROVED** [review]
Thanks, @cpignedoli, looks good!
